[docs] Tweak Material UI's "Template" page design (#37260)
mui/material-ui: docs/data/material/getting-started/templates/templates.md

# React Templates

> <p class="description">A selection of free Material UI templates to help you get started building your app. The collection contains react dashboard, react admin, and more.</p>

Author: I'm unsure with "react dashboard" and "reach admin" here refer to an actual package, thus the lowercase, or if we should capitalize them to React dashboard and React admin instead. 🤔

Author: Exactly! Well, I'm happy to not touch it at this moment but thought of just giving a heads-up. Handing it to you 😬

> <p class="description">A selection of free Material UI templates to help you get started building your app. The collection contains react dashboard, react admin, and more.</p>

Reviewer: How about something like this? I think it keeps the same keywords but makes them work a little better grammatically.

```suggestion
<p class="description">Browse our collection of free React templates to help you get started building your app with Material UI, including a React dashboard, React admin panel, and more.</p>
```

<!-- #default-branch-switch -->

> 

Reviewer: Adding an intro statement here to reiterate the page description and lead into the body text.

```suggestion

Our curated collection of Material UI templates includes dashboards, sign-in and sign-up pages, blog, checkout flow, and more.

```

For multi-part examples, a table in the README at the linked source code location describes
the purpose of each file.

{{"component": "docs/src/modules/components/MaterialFreeTemplatesCollection.js"}}

> If while using these examples you make changes or enhancements that could improve the
> developer experience, or you would like to contribute an additional example,
> please consider creating a [pull request on GitHub](https://github.com/mui/material-ui/pulls).

Reviewer: This can definitely be condensed!

```suggestion
See any room for improvement?
Please feel free to open an [issue](https://github.com/mui/material-ui/issues/new/choose) or [pull request](https://github.com/mui/material-ui/pulls) on GitHub.
```

## Premium templates

> Looking for something more? You can find complete templates & themes in the <a href="https://mui.com/store/?utm_source=docs&utm_medium=referral&utm_campaign=templates-store">premium template section</a>.

Reviewer: ```suggestion
Looking for something more? You can find complete templates and themes in the <a href="https://mui.com/store/?utm_source=docs&utm_medium=referral&utm_campaign=templates-store">premium template section</a>.
```
